### PR TITLE
Add new component: ci-framework

### PR DESCRIPTION
**Pull request by cjeanner**
While it's not an operator per se, it's deeply linked to the install_yamls ecosystem provided by openstack-k8s-operators.

The project will need to run tests upon pull requests, such as pre-commit and molecule testing against ansible roles.

### CODE REVIEW

**Comment by sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-tests

**Comment by cjeanner** [comment]
Note: I've followed the path to get added to the org, things should be OK soon on that side hopefully.

**Comment by sandeepyadav93** [comment]
This looks good, waiting for [1] which add the `make tests` target


[1] https://github.com/openstack-k8s-operators/ci-framework/pull/3

**Comment by cjeanner** [comment]
https://github.com/openstack-k8s-operators/ci-framework/pull/3 just merged - so we should be fine with this one now :).

**Comment by sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-tests

**Comment by cjeanner** [comment]
Hmm I see pyhton-3.6 being used - is it exptected? Did I miss anything anyware? The "permission denied" doesn't seem to be linked to my changes at least...

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-tests

File: ci-operator/config/openstack-k8s-operators/ci-framework/openstack-k8s-operators-ci-framework-master.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+build_root:
```
Comment (sandeepyadav93) [inline ci-operator/config/openstack-k8s-operators/ci-framework/openstack-k8s-operators-ci-framework-master.yaml:1]: Currently, You are using golang-1.13 container as a base container.
~~~
build_root:
  image_stream_tag:
    name: release
    namespace: openshift
    tag: golang-1.13 >>>
~~~

~~~
- as: tests
  commands: make tests
  container:
    from: src  >>>
~~~    



We need to use an image that suits our test requirement, We have a few options.

* Create an image using dockerfile_literal/file:

ex: https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/nova-operator/openstack-k8s-operators-nova-operator-master.yaml#L12-L19

* Reuse an image which already has the deps we need. 

fyi.. We do create an image for our e2e job/precommit job via openstack-k8s-operators-ci[1] if you want to reuse it.

To use it add a .ci-operator.yaml in ci-framerwork repo, example[2]

then you need to add `from_repository: true` and use src as base
~~~
build_root:
  from_repository: true
.
.  
- as: tests
  commands: make tests
  container:
    from: src 
~~~  

If you have to add more deps on src(base container), then you can follow this kind of pattern to add additional deps[3] and use that container[4]


[1] https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/blob/master/images/build-root/Dockerfile.custom
[2] https://github.com/openstack-k8s-operators/openstack-operator/blob/master/.ci-operator.yaml
[3] https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/openstack-ansibleee-operator/openstack-k8s-operators-openstack-ansibleee-operator-main.yaml#L14-L21
[4] https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/openstack-ansibleee-operator/openstack-k8s-operators-openstack-ansibleee-operator-main.yaml#L83-L84

Comment (cjeanner) [reply ci-operator/config/openstack-k8s-operators/ci-framework/openstack-k8s-operators-ci-framework-master.yaml]: Thanks for the help @sandeepyadav93 !

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-tests

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
(sorry for all the noise - but since there's apparently no Depends-On relation, we have to iterate, and iterate, and iterate.... If there's a better way, please share :) )

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/hold

Need to clean things up on the other side, and I want to ensure my cleaning doesn't break this all-green patch :)

**Comment by cjeanner** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit

**Comment by cjeanner** [comment]
/hold cancel

It's working \o/

**Comment by viroel** [comment]
/pj-rehearse ack

pre-commit job is passing: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/36398/rehearse-36398-pull-ci-openstack-k8s-operators-ci-framework-master-pre-commit/1627749672103710720/artifacts/test/build-log.txt

**Pull request merged**